enterprise/frontend/src/embedding-sdk/components/private/InteractiveQuestion/components/ChartTypeSelector.tsx

```diff
@@ -0,0 +1,52 @@
+import {
+  ChartTypeSettings,
+  useChartTypeVisualizations,
+} from "metabase/query_builder/components/chart-type-selector";
+import { Stack } from "metabase/ui";
+
+import { useInteractiveQuestionContext } from "../context";
+
+// Public facing hook so users can create their own selectors for their UI.
+// Will document once this PR is ready to go.
+export const useChartTypeSelectors = () => {
```

Comment:
How will this be used?
We only have the context it needs in our private components right now I think

Reply:
Really good question. I'll do something about it - I was thinking that clients could use it within the customizable InteractiveQuestion if they didn't want to use our buttons, but then they would need to make their own component, so the documentation could become confusing

Reply:
This will allow people to change the visualization from their code right?
Ignoring our current implementation, I think as a user I would expect something like:

```
const usersOverTimeQuestion = useInteractiveQuestion({id:123});

const something = useChartTypeSelector(usersOverTimeQuestion);

return <InteractiveQuestion questionId={usersOverTimeQuestion.id}/>
```

Reply:
But I think until we figure out the external API we can leave out the code from the PR, we don't have many tests yet which means that removing/changing code comes with some friction

Reply:
This is not re-exported yet, so I suppose this hook won't be public-facing yet for this PR's release.